STATE_TRANSITIONS.MD

```diff
@@ -0,0 +1,227 @@
+# Run State Transitions
+Run states change based on the success or failure of a job run. The datasets consumed and/or produced
+by a job run are immutable and do not change based on the success or failure of the run.
+
+# Dataset State Transitions
```

Comment:
Though I agree that datasets have _states_, I think the title can be confused as _states_ is an overloaded term in the context of a run. Mind updating the title to something like: _Dataset Versioning Lifecycle_ or something similar?

Reply:
My thinking was primarily around documenting like a state machine, hence the wording. I think something like _Dataset Versioning Lifecycle_ sounds good, though.